atom/browser/api/atom_api_web_contents.cc

```diff
@@ -789,9 +790,14 @@ void WebContents::AddNewContents(content::WebContents* source,
     if (disposition != WindowOpenDisposition::NEW_WINDOW &&
         disposition != WindowOpenDisposition::NEW_POPUP) {
       if (!browser) {
-        // TODO(bridiver) - this should be the source owner_window
-        browser = owner_window()->browser();
-        tab_helper->SetWindowId(browser->session_id().id());
+        auto relay = atom::NativeWindowRelay::FromWebContents(source);
+        if (relay) {
```

Comment:
I'm pretty sure this doesn't change anything. The native window relay and the owner window are set at exactly the same time https://github.com/brave/muon/blob/master/atom/browser/common_web_contents_delegate.cc#L202
if one of them is null then both of them are null